src/plugins/vis_type_table/README.md

```diff
@@ -1 +1,8 @@
-Contains the data table visualization, that allows presenting data in a simple table format.
\ No newline at end of file
+Contains the data table visualization, that allows presenting data in a simple table format.
+
+By default a new version of visualization will be used. To use the previous version of visualization the config must have the `vis_type_table.legacyVisEnabled: true` setting
+configured in `kibana.dev.yml` or `kibana.yml`, as shown in the example below:
+
```

Comment:
I'm not in favor of keeping the legacy data table available behind a flag.  I would expect when we reactify and EUIify the data table it simply replaces the old angular one.  If you make the old one available behind a flag, then we also have to have telemetry on it to know if users are still using it.  If they're still using it, we'd need to know why.  

Flags like this which require restarting Kibana also are painful for CI jobs.  If we did keep the legacy table behind this flag, I would suggest we stop testing it.

If we're not confident to replace the old data table completely we should wait to merge it until we are confident.

Reply:
We should def add telemetry to this flag to make sure we understand if people disable this. The reasoning behind the flag was, that data table is the 2nd most used visualization across Kibana, and we're fundamentally exchanging the implementation here. I would like to merge it only once we're certain it's 100% working, but in the past we've seen with a lot of those fundamental refactorings, things still broke and even if it's just some edge-cases only "a handful" of users are having. By that measurement I am not sure I would ever feel "confident" enough to replace it without any fallback flag, or rather feel like we're again using a minor version and our user-case to "test" an implementation for us in all cases.

We've also seen that significant refactorings ever so often required some users to switch back to the old implementation (most recently the conversion from _msearch -> _search might have been a good example were we saw quit some users requiring this flag to fall back).

Ideally this flag only lives for a really short time - I would "timebox" it to max. 2 minor versions (ideally only 1). If you're fine with us not having functional tests for the old implementation in place in favor of not requiring another job runner, we could do that. It just felt safer from my perspective to have at least some very basic tests running.

Reply:
Running all listed legacy tests will add extra ~20-25 min, that sounds too much for legacy functionality, we will have to re-group several CI groups to optimize overall execution time.
But I think having a few tests with basic checks (rendering data) is a good compromise (5-7 extra minutes to the same **CI group 9**), we don't want to worry if backward solution is functioning. @sulemanof How about cut some tests from `legacy/data_table/_data_table.ts` ?

Reply:
@dmlemeshko @LeeDr 
I left only few legacy tests to check basic functionality (basic table, few aggs, filtering, table splitting)
CI time is increased with around 5 extra mins -> https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86727/execution/node/531/log/?consoleFull
I believe this is quite enough to check the legacy functionality and is not so painful for CI. What do you think?

Reply:
Personally, I think it is a good way to go.